### PR TITLE
Bug 2038115: Make namespace bar full width and sticky in console

### PR DESCRIPTION
**Pull request by karthikjeeyar**
**Fixes**: 

https://issues.redhat.com/browse/OCPBUGSM-39111

**Analysis / Root cause**: 

The scrollbar is attached to the whole content area in the console.

**Solution Description**: 

Previously, the main content area was scrollable, but we need only the content-scrollable area to be scrollable and the namespace bar which is above the content-scrollable area should occupy full width of the screen.

**Screen shots / Gifs for design review**: 

Before:
Screenshot 2022-01-17 at 4 17 30 PM

After:
Screenshot 2022-01-17 at 4 18 24 PM

https://user-images.githubusercontent.com/9964343/149753920-46a9d1a0-1c5a-484a-83d5-25c4af8dd00f.mov

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

cc: @serenamarie125 @christianvogt

### CODE REVIEW

**Comment by spadgett** [comment]
/assign @rhamilto @sg00dwin 
/hold

@rhamilto @sg00dwin PTAL as this is a fundamental change to the page structure. We also need to test that this doesn't break the ACM plugin before merging. (Adding a hold until we can test.) We should also understand why the original change was made in 4.10 so that we don't regress a different bug.

**Comment by rhamilto** [comment]
> @rhamilto @sg00dwin PTAL as this is a fundamental change to the page structure. We also need to test that this doesn't break the ACM plugin before merging. (Adding a hold until we can test.) We should also understand why the original change was made in 4.10 so that we don't regress a different bug.

@sg00dwin has wrestled with this much more frequently than I have, so I will defer to his review.  IIRC, this change has been implemented and undone more than once (whether intentionally or unintentionally).

**Comment by christianvogt** [comment]
Do we know what change caused the breakage in the first place so we understand what it was trying to solve?

**Comment by karthikjeeyar** [comment]
> Do we know what change caused the breakage in the first place so we understand what it was trying to solve?

Here is the[ PR - 9974](https://github.com/openshift/console/pull/9974)  which changed the page structure to align with PF page structure.

**Comment by sg00dwin** [comment]
> > Do we know what change caused the breakage in the first place so we understand what it was trying to solve?
> 
> Here is the[ PR - 9974](https://github.com/openshift/console/pull/9974) which changed the page structure to align with PF page structure.

The original changes in PR  https://github.com/openshift/console/pull/9974  were to align our page markup structure closer to the default PatternFly page structure to minimize the styling/display issues with the `dynamic-demo-plugin` pages.

https://issues.redhat.com/browse/OCPBUGSM-39111 appeared after [9974](https://github.com/openshift/console/pull/9974) merged.  Where the Developer console `<NamespaceBar>` started scrolling with the page content because its location, within the primary page markup, is nested differently than the Admin console.
https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/components/NamespacedPage.tsx#L31-L39

That being said, I pulled down these pr changes and tested it with the `dynamic-demo-plugin` pages installed and couldn't find any issues when cross browser testing. So it works for me.

**Review by jerolimov — APPROVED** [review]
All pages I tested in admin and dev perspective works fine.

Just saw some edgecases with small screen sizes independent of this change.

@spadgett Wdyt? Can we merge it?

**Comment by jerolimov** [comment]
/retitle Bug 2038115: Make namespace bar full width and sticky in console

File: frontend/public/components/app-contents.tsx
Comment (karthikjeeyar) [reply frontend/public/components/app-contents.tsx]: updated

**Comment by jerolimov** [comment]
Will check this later against ACM with Jon
/retest

**Comment by TheRealJon** [comment]
I've tested this with the ACM dynamic plugin and don't see any major issues. The only problem I see is that ACM uses scrollable areas within it's page content, and the scrollbar can still overlap their static horizontal nav tabs as seen here:

**Comment by TheRealJon** [comment]
This change also makes the "sticky top" page header on the ACM pages work more closely to what's expected:

Before:
screencast-github com-2022 01 28-15_26_07


After:
video-convert

**Comment by jerolimov** [comment]
Thanks Jon!

/bugzilla refresh

**Comment by karthikjeeyar** [comment]
Thanks all for  the reviews.

File: frontend/public/components/app-contents.tsx
Comment (zherman0) [inline frontend/public/components/app-contents.tsx:686]: @karthikjeeyar - FYI: this change deleted the route for the new 'debug pod' feature.  I have created a bug https://bugzilla.redhat.com/show_bug.cgi?id=2053304 and fix at https://github.com/openshift/console/pull/11040

Comment (karthikjeeyar) [reply frontend/public/components/app-contents.tsx:686]: I am extremely sorry about that, not sure how this line got removed (could be due a rebase) and Thanks for finding and fixing it.

**Pull request merged**
